### ISSUE TITLE
When moving lines between tickets: APIError: OrderDetail orderid (0125204OY47A) does not match Order orderid (0125204OY47)

### ISSUE THREAD

**Issue by kbower** [kbower/tickettest1#1399]
_Imported from trac ticket 1399.
Created by **kbower**
Opened in trac: Jan 25, 2012, 8:57:24 AM
Last modified in trac: Jan 26, 2012, 4:49:48 PM_
***
```
File '/home/rarch/trunk/src/appserver/pylotengine/controllers/dbentity.py', line 388 in get_updates
  sa_obj = self.modelClass.convert_to_sa(passed_obj)
File '/home/rarch/trunk/src/appserver/pylotengine/model/objects.py', line 306 in convert_to_sa
  sa_obj = cls._convert_to_sa(dict_obj)
File '/home/rarch/trunk/src/appserver/pylotengine/model/objects.py', line 333 in _convert_to_sa
  sa_obj._set_sa_property(prop, value)
File '/home/rarch/trunk/src/appserver/pylotengine/model/objects.py', line 382 in _set_sa_property
  value = [field_class._convert_to_sa(i) for i in value]
File '/home/rarch/trunk/src/appserver/pylotengine/model/objects.py', line 348 in _convert_to_sa
  sa_obj.check_passed_fks()
File '/home/rarch/trunk/src/appserver/pylotengine/model/objects.py', line 1264 in check_passed_fks
  fks_match(r, i)
File '/home/rarch/trunk/src/appserver/pylotengine/model/objects.py', line 1231 in fks_match
  raise APIError ("%s %s (%s) does not match %s %s (%s)" % (rel_obj.__class__.__name__, r.key, rem_val, self.__class__.__name__, l.key, local_val))
APIError: OrderDetail orderid (0125204OY47A) does not match Order orderid (0125204OY47)
```

**Comment by kbower** [comment]
_**kbower** trac comment on Jan 26, 2012, 4:49:21 PM:_

(only happens directly after saving new order.  the reason is movedfromorderuid is being sent back from the server even after a save, after which the orderdetails pk has been updated)

**Comment by kbower** [comment]
_**kbower** trac comment on Jan 26, 2012, 4:49:48 PM:_

(Changeset [[2232]](../commit/1b911b464bd8a9f01ca18a1e819e3cdb9f9cbb0e))
 * fixes #1399 (When moving lines between tickets: APIError: OrderDetail orderid (0125204OY47A) does not match Order orderid (0125204OY47))

**Comment by kbower** [comment]
_**kbower** trac status change on Jan 26, 2012, 4:49:48 PM:_ **closed**
